internal/provider/security_context.go

```diff
@@ -9,9 +9,12 @@ import (
 )
 
 // uidGidFromSecurityContext returns the uid and gid (as a string) from the SecurityContext.
-// If windowsOptions are set, the uid and gid *names* found there are returned. This takes
-// precedence over the values runAsUser and runAsGroup.
+// If windowsOptions are set, the uid and gid *names* found there are returned. If these users
```

Comment:
Why are we caring about `windowsOptions`? As per docs:
>The Windows specific settings applied to all containers. If unspecified, the options within a container's SecurityContext will be used. If set in both SecurityContext and PodSecurityContext, the value specified in SecurityContext takes precedence.

Reply:
I actually just realized our code covers this for a while now 😕

Reply:
yes, and you def want to care because I want to run my unit as user 'prometheus', not some rando UID which now clashes with the debian package.

Reply:
Not sure I follow. `windowsOptions` are specific to Windows. There's no Windows support here so what am I missing?

Reply:
there is no other way to define a 'username' to run your pod as, so this is windows in name only, and def. a must have feature in this case. Otherwise it would have to be a annotation or some such to define the user (and potentially the group) to run the unit as.

Reply:
Ah I get it now.

Reply:
Still, I don't see many Linux users actually doing this or understanding systemk can do this because of the PodSpec documentation that I quoted above. So maybe add some docs?